### PR TITLE
Update and fix Aqua test

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR updates and fixes an aqua test.

### CODE REVIEW

File: test/aqua.jl [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
-import ClimaCore
+using ClimaCore
 using Aqua
 
-@testset "Aqua tests - unbound args" begin
+@testset "Aqua tests (performance)" begin
     # This tests that we don't accidentally run into
     # https://github.com/JuliaLang/julia/issues/29393
-    Aqua.test_unbound_args(ClimaCore)
+    # Aqua.test_unbound_args(ClimaCore)
```
Comment (juliasloan25) [inline test/aqua.jl:8]: Should this line be kept as a comment or just removed?

File: test/aqua.jl [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
 
-    # returns a vector of all unbound args
-    # ua = Aqua.detect_unbound_args_recursively(ClimaCore)
+@testset "Aqua tests (additional)" begin
+    Aqua.test_undefined_exports(ClimaCore)
+    Aqua.test_stale_deps(ClimaCore)
+    Aqua.test_deps_compat(ClimaCore)
+    Aqua.test_project_extras(ClimaCore)
+    # Aqua.test_project_toml_formatting(ClimaCore) # failing
+    Aqua.test_piracy(ClimaCore) # failing
```
Comment (juliasloan25) [inline test/aqua.jl:32]: Is this still failing? It looks like all the tests pass on both buildkite and GHA

File: test/aqua.jl [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
     ambs = Aqua.detect_ambiguities(ClimaCore; recursive = true)
     pkg_match(pkgname, pkdir::Nothing) = false
     pkg_match(pkgname, pkdir::AbstractString) = occursin(pkgname, pkdir)
     filter!(x -> pkg_match("ClimaCore", pkgdir(last(x).module)), ambs)
-    for method_ambiguity in ambs
-        @show method_ambiguity
-    end
-    # If the number of ambiguities is less than the limit below,
-    # then please lower the limit based on the new number of ambiguities.
-    # We're trying to drive this number down to zero to reduce latency.
```
Comment (juliasloan25) [inline test/aqua.jl:21]: It might be helpful to keep this comment (or some version of it)

**Review by juliasloan25 — COMMENTED** [review]
Looks good! I just had a few comments

File: test/aqua.jl (hunk 1 above)
Comment (charleskawczynski) [reply test/aqua.jl:8]: Yes, we can just remove it

File: test/aqua.jl (hunk 2 above)
Comment (charleskawczynski) [reply test/aqua.jl:32]: Ah, right, I forgot to remove-- I think I recently fixed it, actually

File: test/aqua.jl [hunk 4]
```diff
@@ -1,27 +1,37 @@
...
+end
 
-    # returns a vector of all unbound args
-    # ua = Aqua.detect_unbound_args_recursively(ClimaCore)
+@testset "Aqua tests (additional)" begin
+    Aqua.test_undefined_exports(ClimaCore)
+    Aqua.test_stale_deps(ClimaCore)
+    Aqua.test_deps_compat(ClimaCore)
+    Aqua.test_project_extras(ClimaCore)
+    # Aqua.test_project_toml_formatting(ClimaCore) # failing
```
Comment (juliasloan25) [inline test/aqua.jl:33]: Is this line still failing?

Comment (charleskawczynski) [reply test/aqua.jl:33]: Yes

**Pull request merged**